mava/systems/tf/maddpg/training.py

```diff
@@ -451,36 +457,43 @@ def _backward(self) -> None:
         for agent in self._trainer_agent_list:
             agent_key = self._agent_net_keys[agent]
 
+            if self._use_single_optimizer:
             # Get trainable variables.
-            policy_variables = (
-                self._observation_networks[agent_key].trainable_variables
-                + self._policy_networks[agent_key].trainable_variables
-            )
-            critic_variables = (
-                # In this agent, the critic loss trains the observation network.
-                self._observation_networks[agent_key].trainable_variables
-                + self._critic_networks[agent_key].trainable_variables
-            )
+                policy_variables = (
+                    self._observation_networks[agent_key].trainable_variables
+                    + self._policy_networks[agent_key].trainable_variables
+                    + self._critic_networks[agent_key].trainable_variables
+                )
+            else:
+                policy_variables = (
+                    self._observation_networks[agent_key].trainable_variables
```

Comment:
I think this was broken before, but like acme https://github.com/deepmind/acme/blob/b9681d9d9ad9ccaec4271bf2abeb25281185def2/acme/agents/tf/ddpg/learning.py#L215 , we should only include obs vars as part of the critic vars, not policy ones.